lib/core/src/Cardano/Wallet.hs

```diff
@@ -1441,9 +1582,25 @@ selectAssets
     -> (s -> Selection -> result)
     -> ExceptT ErrSelectAssets IO result
 selectAssets ctx params transform = do
-    guardPendingWithdrawal
     pp <- liftIO $ currentProtocolParameters nl
-    liftIO $ traceWith tr $ MsgSelectionStart
+    selectAssets' tr tl pp params transform
+  where
+    nl = ctx ^. networkLayer
+    tl = ctx ^. transactionLayer @k
+    tr = contramap MsgWallet $ ctx ^. logger
+
+-- | Less convenient and restrictive version of `selectAssets`.
+selectAssets'
+    :: forall m s k result. MonadRandom m
+    => Tracer m WalletLog
+    -> TransactionLayer k SealedTx
+    -> ProtocolParameters
+    -> SelectAssetsParams s result
```

Comment:
Would it be helpful to add these other parameters into the `SelectAssetsParams` record type?